fars/booking/tests.py

```diff
@@ -1,3 +1,114 @@
-from django.test import TestCase
+from django.test import TestCase, Client
+from booking.models import Bookable, Booking
+from django.contrib.auth import get_user_model
+from datetime import datetime, timedelta
 
 # Create your tests here.
+class BookTestCase(TestCase):
+    def setUp(self):
+        Bookable.objects.create(
+            id_str = "b",
+            name = "B",
+            description = "A bookable"
+        )
+
+        User = get_user_model()
+        User.objects.create_user(
+            username = "u",
+            password = "pw"
+        )
+
+
+    def test_user_can_book(self):
+        """A valid user can book a bookable"""
+        c = Client()
+        now = datetime.now()
+        postdata = {
+            'before': now.strftime("%Y-%m-%dT%H:%M:%SZ"),
+            'after': (now + timedelta(hours=1)).strftime("%Y-%m-%dT%H:%M:%SZ"),
+            'bookable': 'b'
+        }
+
+        c.login(username='u', password='pw')
```

Comment:
And this is why you want to create your test data inside your tests. You'd want to avoid using these hardcoded strings.